modules/platform/nuxeo-platform-csv-export/src/main/java/org/nuxeo/ecm/platform/csv/export/io/DocumentPropertyCSVWriter.java

```diff
@@ -168,17 +174,25 @@ protected String getTranslatedValue(Session session, String schema, String lang,
         } catch (PropertyNotFoundException e) {
             try {
                 // Check if it comes from a l10nxvocabulary, and return this value if it exists, or else return the id
-                return (String) entry.getProperty(schema, "label_en");
+                return replaceCarriageReturn((String) entry.getProperty(schema, "label_en"));
             } catch (PropertyNotFoundException e1) {
-                return value;
+                return replaceCarriageReturn(value);
             }
         }
         Locale locale = lang != null ? Locale.forLanguageTag(lang) : ctx.getLocale();
-        return I18NUtils.getMessageString("messages", label, new Object[0], locale);
+        return replaceCarriageReturn(I18NUtils.getMessageString("messages", label, new Object[0], locale));
     }
 
     protected void writeUnsupported(Type type, CSVPrinter printer) throws IOException {
         printer.print(String.format("type %s is not supported", type.getName()));
     }
 
+    protected String replaceCarriageReturn(String value) {
+        String replacement = ctx.getParameter(CARRIAGE_RETURN_REPLACEMENT_CTX_DATA);
+        if (value == null || replacement == null) {
+            return value;
+        }
+        return value.replace(System.lineSeparator(), replacement);
```

Comment:
I'd prefer something more general as a file with one kind of newline may be treated on a platform with another kind.

    protected static final Pattern NEWLINE = Pattern.compile("\\R");

    NEWLINE.matcher(value).replaceAll(replacement);

Reply:
Thanks for the code 👍